### PR TITLE
SDIT-1929 Endpoint to get basic prisoner details for the latest booking

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt [hunk 1]
```diff
@@ -24,10 +25,8 @@ class PrisonerService(
   private val mergeTransactionRepository: MergeTransactionRepository,
   private val offenderBookingRepository: OffenderBookingRepository,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt:26]: ktlint converted these to expression bodies

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt [hunk 2]
```diff
@@ -71,3 +67,10 @@ class PrisonerService(
       ?.let { PreviousBookingId(bookingId = it.bookingId, bookingSequence = it.bookingSequence!!.toLong()) }
       ?: throw NotFoundException("Prisoner with offenderNo $offenderNo and booking $bookingId not found or has no previous booking")
 }
+
+private fun OffenderBooking.toPrisonerDetails(): PrisonerDetails = PrisonerDetails(
+  offenderNo = offender.nomsId,
+  bookingId = bookingId,
+  location = location?.id ?: "",
+  active = active,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt:75]: new field here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt [hunk 3]
```diff
@@ -45,24 +44,21 @@ class PrisonerService(
   fun findAllPrisoners(pageRequest: Pageable): Page<PrisonerId> =
     offenderRepository.findAllIds(pageRequest).map { PrisonerId(offenderNo = it.getNomsId()) }
 
-  fun findPrisonerDetails(bookingIds: List<Long>): List<PrisonerDetails> {
-    return bookingRepository.findAllById(bookingIds)
-      .map { PrisonerDetails(it.offender.nomsId, it.bookingId, it.location?.id ?: "") }
-  }
+  fun findPrisonerDetails(bookingIds: List<Long>): List<PrisonerDetails> = bookingRepository.findAllById(bookingIds)
+    .map { it.toPrisonerDetails() }
+  fun findPrisonerDetails(offenderNo: String): PrisonerDetails = bookingRepository.findLatestByOffenderNomsId(offenderNo)?.toPrisonerDetails() ?: throw NotFoundException("No prisoner with offender $offenderNo found")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisoners/PrisonerService.kt:49]: new function here

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**